R/calc_aggregate_counts.R

```diff
@@ -60,17 +88,45 @@ calc_aggregate_counts <- function(
             collapse_vaccine,
             mutate(., Residents.Vadmin = ifelse(
                 is.na(.$Residents.Vadmin), .$Residents.Completed, .$Residents.Vadmin)),
-            .) %>%
-        tidyr::pivot_longer(
-            -(Name:State), names_to = "Measure", values_to = "UCLA") %>%
-        filter(!is.na(UCLA)) %>%
-        group_by(State, Measure) %>%
-        mutate(has_statewide = "STATEWIDE" %in% Name) %>%
-        # if state wide and other counts exist for a measure only use statewide
-        filter(!(has_statewide & Name != "STATEWIDE")) %>%
-        group_by(State, Measure) %>%
-        summarise(UCLA = sum_na_rm(UCLA), .groups = "drop") %>%
-        full_join(mp_data, by = c("State", "Measure")) %>%
+            .)
+
+    if(all_dates){
+        state_df <- state_wide_df %>%
+            mutate(Date = lubridate::floor_date(Date, round_)) %>%
+            tidyr::pivot_longer(
+                -(Name:State), names_to = "Measure", values_to = "UCLA") %>%
+            filter(!is.na(UCLA)) %>%
+            group_by(State, Date, Measure, Name) %>%
+            summarize(UCLA = max_na_rm(UCLA), .groups = "drop_last") %>%
+            mutate(has_statewide = "STATEWIDE" %in% Name) %>%
+            # if state wide and other counts exist for a measure only use statewide
+            filter(!(has_statewide & Name != "STATEWIDE")) %>%
+            group_by(State, Date, Measure) %>%
+            summarise(UCLA = sum_na_rm(UCLA), .groups = "drop")
+
+        comb_df <- state_df %>%
+            full_join(mp_data, by = c("State", "Measure", "Date"))
+    }
+    else{
+        state_df <- state_wide_df %>%
+            tidyr::pivot_longer(
+                -(Name:State), names_to = "Measure", values_to = "UCLA") %>%
+            filter(!is.na(UCLA)) %>%
+            group_by(State, Measure) %>%
+            mutate(has_statewide = "STATEWIDE" %in% Name) %>%
+            # if state wide and other counts exist for a measure only use statewide
+            filter(!(has_statewide & Name != "STATEWIDE")) %>%
+            group_by(State, Measure) %>%
+            summarise(
+                UCLA = sum_na_rm(UCLA), Date = max(Date), .groups = "drop")
+
+        comb_df <- state_df %>%
+            rename(Date.UCLA = Date) %>%
+            full_join(
+                rename(mp_data, Date.MP = Date), by = c("State", "Measure"))
+    }
+
+    harm_df <- comb_df %>%
         mutate(Val = case_when(
             is.na(UCLA) & is.na(MP) ~ NA_real_,
             is.na(UCLA) ~ MP,
```

Comment:
Should there be a case here for when MP is NA but UCLA is not NA?

Reply:
Yup yup that is [here](https://github.com/uclalawcovid19behindbars/behindbarstools/blob/aggregate_counts/R/calc_aggregate_counts.R#L129-L135). Wasnt captured in the diff for some reason.